analytics_data_api/v0/documents.py

```diff
@@ -160,7 +160,12 @@ def get_users_in_course(
             }
             for sort_policy in sort_policies
         ])
-        return search_request.execute()
+
+        # calling count will generate an additional request to ES, but is necessary for returning the entire set of data
+        total = search_request.count()
+        # by default, ES will only return the first 10 documents, we want to get all of them
+        total_search_requests = search_request[0:total]
```

Comment:
Spent some time looking up the library to make sure slicing like this was kosher, and pagination is the documented use case for slicing like this! Nice find!

It seems like we may want to actually paginate these queries though, rather than (correct me if I'm wrong & this isn't the setup:) pulling in all results from ES and then doing pagination on the result in python code. This approach seems like for large courses it may be really really unperformant. I wouldn't hesitate to say "let's just follow up on that in a separate ticket", but I'm worried large courses which are currently limited to 10 learners might end up in a state where they can't see any since results take a long time to come back. Might be a regression for some courses. Can we test that? Are there large enough courses on staging to replicate that sort of issue?

Reply:
In discussion on slack it seems like we'll test on stage with a large course.

I'd like a follow-up ticket to explore paginating the ES queries themselves, since I think that'll be valuable either way.

Reply:
Great, here: https://openedx.atlassian.net/browse/MST-1056

Reply:
To add, we will test on stage for a course with ~28,000 students. I also created https://openedx.atlassian.net/browse/MST-1056 to explore the ES pagination question.